analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/FinallyPointTest.cs

```diff
@@ -0,0 +1,68 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Extensions;
+using SonarAnalyzer.SymbolicExecution.Roslyn;
+using SonarAnalyzer.UnitTest.Helpers;
+
+namespace SonarAnalyzer.UnitTest.SymbolicExecution.Roslyn
+{
+    [TestClass]
+    public class FinallyPointTest
+    {
+        [TestMethod]
+        public void Constructor_Null_Throws() =>
+            ((Func<FinallyPoint>)(() => new FinallyPoint(null, null))).Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("branch");
```

Comment:
It throws only if the second parameter is null. I think we should highlight that in the test.